### PR TITLE
JSON MQTT topic documentation

### PR DESCRIPTION
**Pull request by michaelkleinhenz**
This adds documentation for https://github.com/meshtastic/Meshtastic-device/pull/1283.

### CODE REVIEW

File: docs/software/other/mqtt.md [hunk 1]
```diff
@@ -35,6 +36,44 @@ The payload is a raw protobuf. Looking at the MQTT traffic with a program like `
...
+   "to":-1,
+   "type":"nodeinfo"
+}
+```
+
+If the message received contains valid JSON in the payload, the JSON is deserialized and added as a JSON object rather than a string containing the serialized JSON.
+
+**Sent messages** will be checked if the MQTT payload contains a valid JSON-encoded envelope:
+
+```
```
Comment (sachaw) [inline docs/software/other/mqtt.md:64]: Please specify language for code blocks: ```json

File: docs/software/other/mqtt.md [hunk 2]
```diff
@@ -35,6 +36,44 @@ The payload is a raw protobuf. Looking at the MQTT traffic with a program like `
 	ShortFast	!937bed1c
 ```
 
+Packets of the types `TEXT_MESSAGE_APP`, `ENVIRONMENTAL_MEASUREMENT_APP`, `NODEINFO_APP` and `POSITION_APP` are encoded to JSON messages and sent to the `msh/1/json/CHANNELID/DEVICEID` topic. 
+
+An example of a received `NODEINFO_APP` message:
+```
```
Comment (sachaw) [inline docs/software/other/mqtt.md:42]: Please specify language for code blocks: ```json

File: docs/software/other/mqtt.md [hunk 3]
```diff
@@ -35,6 +36,44 @@ The payload is a raw protobuf. Looking at the MQTT traffic with a program like `
 	ShortFast	!937bed1c
 ```
 
+Packets of the types `TEXT_MESSAGE_APP`, `ENVIRONMENTAL_MEASUREMENT_APP`, `NODEINFO_APP` and `POSITION_APP` are encoded to JSON messages and sent to the `msh/1/json/CHANNELID/DEVICEID` topic.
```
Comment (sachaw) [inline docs/software/other/mqtt.md:39]: I'd change to wording to something along the lines of:

Messages from the following [port numbers](https://meshtastic.org/docs/developers/firmware/portnum) are serialized to JSON then forwarded to the `msh/1/json/CHANNELID/DEVICEID` topic:
`TEXT_MESSAGE_APP`, `ENVIRONMENTAL_MEASUREMENT_APP`, `NODEINFO_APP` and `POSITION_APP`

File: docs/software/other/mqtt.md [hunk 4]
```diff
@@ -11,16 +11,17 @@ Meshtastic devices with wifi hardware (ESP32) are able to connect to an MQTT bro
...
 - Using a custom MQTT broker to bridge several mesh networks together, via the internet (or just a local IP network)
 - Using a custom MQTT broker and a translator program to decode the raw protobuf packets and translate them into a plain text form for use in other systems. eg plotting temperature readings in Grafana, or device positions in Traccar.
+- Using or emitting packets directly in/from smart home control software such as Home Assistant or other consumers that can work with JSON messages.
 
-When MQTT enabled, the Meshtastic device simply uplinks and/or downlinks every raw protobuf packet that it sees to the MQTT broker. All packets are sent to the broker, whether they originate from another device on the mesh, or the gateway node itself.
+When MQTT enabled, the Meshtastic device simply uplinks and/or downlinks every raw protobuf packet that it sees to the MQTT broker. In addition, some packet types are decoded or encoded from/to JSON messages for easier use in consumers. All packets are sent to the broker, whether they originate from another device on the mesh, or the gateway node itself.
```
Comment (sachaw) [inline docs/software/other/mqtt.md:16]: Please use Serialize/Deserialize vs encode/decode (just more accurate)

File: docs/software/other/mqtt.md [hunk 5]
```diff
@@ -11,16 +11,17 @@ Meshtastic devices with wifi hardware (ESP32) are able to connect to an MQTT bro
...
 
 Packets may be encrypted. If you use the default meshtastic MQTT server, packets are always encrypted. If you use a custom MQTT broker (ie set `mqtt_server`), the `mqtt_encryption_enabled` setting applies, which by default is false.
 
 IMPORTANT: When MQTT is turned on, you are potentially broadcasting your entire mesh traffic onto the public internet. This includes messages and position information.
 
-### MQTT Topic
+### MQTT Topics
 
-The device will uplink and downlink packets to the `msh/` prefix:
+The device will uplink and downlink raw packets to the `msh/` prefix:
```
Comment (sachaw) [inline docs/software/other/mqtt.md:24]: maybe:

The device will uplink and downlink raw([protobuf](https://developers.google.com/protocol-buffers)) packets to the `msh/` prefix:

**Review by sachaw — CHANGES_REQUESTED** [review]
Just a few minor changed, great work otherwise.

File: docs/software/other/mqtt.md (hunk 4 above)
Comment (michaelkleinhenz) [reply docs/software/other/mqtt.md]: Fixed in #6dd1de3.

File: docs/software/other/mqtt.md (hunk 3 above)
Comment (michaelkleinhenz) [reply docs/software/other/mqtt.md]: Fixed in #6dd1de3.

File: docs/software/other/mqtt.md (hunk 1 above)
Comment (michaelkleinhenz) [reply docs/software/other/mqtt.md]: Fixed in #6dd1de3.

**Pull request merged**